internal/metrics/system_test.go

```diff
@@ -0,0 +1,203 @@
+package metrics_test
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/jakub-dzon/k4e-device-worker/internal/metrics"
+	"github.com/jakub-dzon/k4e-operator/models"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"time"
+)
+
+var _ = Describe("System", func() {
```

Comment:
We should test double-update if add-workload is called again or not.

Reply:
Done